### PR TITLE
Update with master

### PR DESCRIPTION
**Pull request by naglepuff**
Bring the `girder-5` branch up to date with changes from `master`

### CODE REVIEW

File: histomicsui/handlers.py
Comment (naglepuff) [inline histomicsui/handlers.py:1]: @manthey @zachmullen 

This file seems to have changed a lot between `master` and `girder-5`.  It looks like `process_annotations` was replaced by `process_annotations_task`, and `process_annotations` has gotten some updates in `master`, checking for file size from the config. (`process_annotations_task` seems to read the entire file in chunks no matter what).

Should I port the changes from `master` over to `process_annotations_task`? (i.e. `import large_image.config`, and copy over the logic?)

File: histomicsui/rest/system.py [hunk 1]
```diff
@@ -15,9 +15,9 @@
 #############################################################################
 
 import datetime
+import logging
```
Comment (naglepuff) [inline histomicsui/rest/system.py:18]: @zachmullen after looking at the differences between `girder-5` and `master`, it looks to me like we're no longer using `girder.logger` and instead importing `logging` and using `getLogger`. Is this correct?

Comment (zachmullen) [reply histomicsui/rest/system.py:18]: Yes, that's correct. I'd recommend reading through this comprehensive list of Girder 5's key changes: https://github.com/girder/girder/blob/v4-integration/docs/migration-guide.rst#3x--5x

File: histomicsui/handlers.py
Comment (naglepuff) [reply histomicsui/handlers.py:1]: @manthey @zachmullen 

I decided to move the changes over in [dc43dfc](https://github.com/DigitalSlideArchive/HistomicsUI/pull/425/commits/dc43dfc61ec70b4c0b554ee70e24174b4cd8efff). It looks like `large_image.config.getConfig` can already work with [environment variables](https://github.com/girder/large_image/blob/7f51c42f7055bac6c369f279c212ee8259d632ff/large_image/config.py#L71-L94).

If there are any objections let me know, I can revert the commit.

**Comment by naglepuff** [comment]
@zachmullen @manthey I'd like input from you both regarding b691b9a, since I feel like I'm missing some context about some of the changes made - both in girder/girder v5, and in the current HistomicsUI code base.

The commit mentioned above wraps some code in an event handler to delay it running until `girder.rest.getApiRoot()` will return a real (i.e. not `undefined`) value. The wrapped code was introduced in PR #429. This code calls `HuiSettings.getSettings`, which in turn calls `girder.rest.restRequest`, which calls `getApiRoot` when building the URL for the rest request.

The issue I was seeing in the girder 5 version of the stack was that at the time this code ran (it's a top-level function in the file, so I believe it would run whenever it was imported), `getApiRoot` returns `undefined`.

I spoke with @manthey and then compared Girder 3 and Girder 5's `rest.js` code. It looks like in v3, there's additional code that sets `apiRoot` directly in `rest.js` (either pulling a value out of an HTML tag with jquery, or using a default value). [See here.](https://github.com/girder/girder/blob/5d1d802a69980988a8c118cd949e31b901fec6a8/girder/web_client/src/rest.js#L37-L39) This code is not present in the `v4-integration` [branch](https://github.com/girder/girder/blob/v4-integration/girder/web/src/rest.js), and I'm not sure exactly why it was removed.

Girder does call `setApiRoot` during the initialization process, but that is [async](https://github.com/girder/girder/blob/db952bea96b162fd9558c76bde5a4759f74e21c3/girder/web/src/main.ts#L17-L46), so my best guess is that code in HistomicsUI is being run before the Girder app has finished initializing (and specifically, before it has got to [this point](https://github.com/girder/girder/blob/db952bea96b162fd9558c76bde5a4759f74e21c3/girder/web/src/index.ts#L26)). In that sense, I think wrapping the code in an event handler makes sense.

If my understanding of things is correct, then there is a difference between Girder 3 and Girder 5 in that plugins being developed for Girder 3 could make `restRequest`s before the Girder app itself had finished initializing, and in Girder 5, plugins can no longer make that assumption.

Maybe this is a side-effect of getting rid of the `build` command, or not, I'm not so sure. But if there are plugins that will break due to this difference maybe this is worth documenting in the migration guide? Am I off base here @zachmullen?

**Pull request merged**